protocols/noise/src/io.rs

```diff
@@ -55,12 +56,48 @@ impl Buffer {
     }
 }
 
+/// A passthrough enum for the two kinds of state machines in `snow`
```

Comment:
Thank you!!!!  I was stuck on this.

Reply:
Yeah, this blocked me for a while too. I tried creating a trait for it, but ran into all kinds of trait bounds problems.